frontend/simple/main.js

```diff
@@ -5,18 +5,27 @@ import VeeValidate from 'vee-validate'
 import SignUp from './views/SignUp.vue'
 import CreateGroup from './views/CreateGroup.vue'
 import UserProfileView from './views/UserProfileView.vue'
+import TestEventLog from './views/EventLog.vue'
 // import NewIncomeView from './views/NewIncomeView.vue'
 import PayGroupView from './views/PayGroupView.vue'
 import NavBar from './views/NavBar.vue'
 import utils, { wrap, lazyLoadVue, superagentHeader } from './js/utils'
 import store from './js/state'
 import './js/transitions'
+import Primus from './assets/vendor/primus'
 
 Vue.use(Router)
 Vue.use(VeeValidate)
 
 superagentHeader('Authorization', `gi ${utils.sign('hello', utils.keypair)}`)
 
+var primus = new Primus(process.env.API_URL, {timeout: 3000, strategy: ['disconnect', 'online', 'timeout']})
+primus.on('disconnection', () => store.commit('updateSocket', null))
+primus.on('error', err => console.log(err))
+primus.on('data', msg => store.dispatch('receiveEvent', msg))
+primus.on('open', () => store.commit('updateSocket', primus))
+primus.on('reconnected', () => store.commit('updateSocket', primus))
+
```

Comment:
Since the pubsub is likely to get quite involved, and since it's good to keep the backend and frontend as similar as possible in structure, let's put this stuff into its own pubsub.js file.